homeassistant/components/climate/systemair_savecair.py

```diff
@@ -0,0 +1,269 @@
+        """Return the list of supported features."""
+        return self._support_flags
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._name
+
+    @property
+    def supply_air_fan_speed(self):
+        """Return the supply air fan speed."""
+        if "digital_input_tacho_saf_value" not in self._client.data:
+            return None
+
+        return self._client.data["digital_input_tacho_saf_value"]
+
+    @property
+    def extract_air_fan_speed(self):
+        """Return the extract air fan speed."""
+        if "digital_input_tacho_eaf_value" not in self._client.data:
+            return None
+
+        return self._client.data["digital_input_tacho_eaf_value"]
+
+    @property
+    def supply_air_temperature(self):
+        """Return the supply air temperature."""
+        if "supply_air_temp" not in self._client.data:
+            return None
+
+        air_temperature = self._client.data["supply_air_temp"]
+
+        if air_temperature is None:
+            return None
+
+        return air_temperature / 10
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        if "control_regulation_temp_unit" not in self._client.data:
+            return TEMP_CELSIUS
+
+        if self._client.data["control_regulation_temp_unit"] == "celsius":
+            return TEMP_CELSIUS
+
+        return TEMP_FAHRENHEIT
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if "pdm_input_temp_value" not in self._client.data:
+            return None
+
+        current_temperature = self._client.data["pdm_input_temp_value"]
+
+        if current_temperature is None:
+            return None
+
+        return current_temperature / 10
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if "main_temperature_offset" not in self._client.data:
+            return None
+
+        target_temperature = self._client.data["main_temperature_offset"]
+
+        if target_temperature is None:
+            return None
+
+        return target_temperature / 10
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        if "rh_sensor" not in self._client.data:
+            return None
+
+        return self._client.data["rh_sensor"]
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        if "main_airflow" not in self._client.data:
+            return None
+
+        if self._client.data["main_airflow"] is None:
+            return None
+
+        fan_mode = int(self._client.data["main_airflow"])
+        ha_fan_mode = SAVECAIR_FAN_MODES[fan_mode]
+
+        return ha_fan_mode
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self._client.get_current_operation()
```

Comment:
This should return one of the climate base component states. If possible create a persistent dict that maps client mode to home assistant state lookup the correct state here.

Create the persistent dict in init, or if we don't need to import anything from the client library to create the dict, we can put it at the module level.